File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Developing Starter Master Pages/C#/WCM_StarterMasterPage/Features/StarterMasterPageFeature/StarterMasterPageFeature.EventReceiver.cs
using System;
using System.Runtime.InteropServices;
using System.Security.Permissions;
using Microsoft.SharePoint;
using Microsoft.SharePoint.Security;

namespace WCM_StarterMasterPage.Features.StarterMasterPageFeature
{
    /// <summary>
    /// This feature receiver sets the master page of the current site to be the
    /// master page deployed by the StarterMasterPageModule.
    /// </summary>

    [Guid("ae8d05aa-da52-4a45-8915-b0b79c4f0d66")]
    public class StarterMasterPageFeatureEventReceiver : SPFeatureReceiver
    {

        public override void FeatureActivated(SPFeatureReceiverProperties properties)
        {
            //Get the current web site, ensuring proper disposal
            using (SPWeb currentWeb = (SPWeb)properties.Feature.Parent)
            {
                //Set the master page values and update the SPWeb
                currentWeb.MasterUrl = "/_catalogs/masterpage/_starter_publishing.master";
                currentWeb.CustomMasterUrl = "/_catalogs/masterpage/_starter_publishing.master";
                currentWeb.Update();
            }
        }

        public override void FeatureDeactivating(SPFeatureReceiverProperties properties)
        {
            //We must set the master page back to the default: v4.master
            using (SPWeb currentWeb = (SPWeb)properties.Feature.Parent)
            {
                currentWeb.MasterUrl = "/_catalogs/masterpage/v4.master";
                currentWeb.CustomMasterUrl = "/_catalogs/masterpage/v4.master";
                currentWeb.Update();
            }
        }

    }
}

File: Microsoft Office Developer Documentation Team/SharePoint 2010 Developing Starter Master Pages/[VB]-SharePoint 2010 Developing Starter Master Pages/VB/WCM_StarterMasterPage/Features/StarterMasterPageFeature/StarterMasterPageFeature.EventReceiver.vb
Option Explicit On
Option Strict On

Imports System
Imports System.Runtime.InteropServices
Imports System.Security.Permissions
Imports Microsoft.SharePoint
Imports Microsoft.SharePoint.Security

''' <summary>
''' This feature receiver sets the master page of the current site to be the
''' master page deployed by the StarterMasterPageModule.
''' </summary>

<GuidAttribute("60e066dd-0263-41e8-89ad-fc6b9b32bc5b")> _
Public Class StarterMasterPageFeatureEventReceiver 
    Inherits SPFeatureReceiver

    Public Overrides Sub FeatureActivated(ByVal properties As SPFeatureReceiverProperties)
        'Get the current web site, ensuring proper disposal
        Using currentWeb As SPWeb = TryCast(properties.Feature.Parent, SPWeb)
            'Set the master page values and update the SPWeb
            currentWeb.MasterUrl = "/_catalogs/masterpage/_starter_publishing.master"
            currentWeb.CustomMasterUrl = "/_catalogs/masterpage/_starter_publishing.master"
            currentWeb.Update()
        End Using
    End Sub

    Public Overrides Sub FeatureDeactivating(ByVal properties As SPFeatureReceiverProperties)
        'We must set the master page back to the default: v4.master
        Using currentWeb As SPWeb = TryCast(properties.Feature.Parent, SPWeb)
            currentWeb.MasterUrl = "/_catalogs/masterpage/v4.master"
            currentWeb.CustomMasterUrl = "/_catalogs/masterpage/v4.master"
            currentWeb.Update()
        End Using
    End Sub

End Class
